src/NuGet.Core/NuGet.Common/ConcurrencyUtilities.cs

```diff
@@ -251,15 +252,19 @@ private static string FilePathToLockName(string filePath)
             // the ctor of semaphore looks for the file and throws an IOException
             // when the file doesn't exist. So we need a conversion from a file path
             // to a unique lock name.
-            using (var sha = SHA1.Create())
+            using (var sha = SHA256.Create())
             {
                 // To avoid conflicts on package id casing a case-insensitive lock is used.
                 var fullPath = Path.IsPathRooted(filePath) ? Path.GetFullPath(filePath) : filePath;
                 var normalizedPath = fullPath.ToUpperInvariant();
 
                 var hash = sha.ComputeHash(Encoding.UTF32.GetBytes(normalizedPath));
 
-                return ToHex(hash);
+                //Changing from SHA-1 to SHA-256 would increase hex hash string by 24 characters (160 bits to 256 bits).
+                //In order not to hit the MAX_PATH limitation, truncate the hash byte array length from 32 to 20.
+                byte[] truncatedHash = new byte[HashLength];
```

Comment:
If you're truncating 256 to 160 bit then what is actual benefit of moving to SHA-256?
Are you trying to match another algorithm somewhere else?

Reply:
SHA1 is not secure so it's not allowed to use anymore.
There're some detailed instructions in the https://github.com/NuGet/Client.Engineering/issues/449

Reply:
But truncating it, we're reducing effectiveness hash function. Isn't it?

Reply:
Yes, truncating a SHA-256 to 160 bits reduces its effectiveness, but it is still preferred over SHA-1.  See SDL requirements for details.